packages/reports/addon/components/dimension-bulk-import.js

```diff
@@ -111,6 +120,19 @@ export default Ember.Component.extend({
     set(this, '_loadingPromise', DS.PromiseObject.create({ promise }));
   },
 
+  /**
+   * Gets the searchable id field if dimension has an 'id' tag, primaryKey otherwise
+   * @private
+   * @method _searchableIdField
+   * @returns {String} - Field to search against
+   */
+  _searchableIdField() {
+    const meta = get(this, '_bardMetadata').getById('dimension', get(this, 'dimension.name'));
+    //see if this field has a tagged id field
+    const field = meta.getFieldsForTag('id')[0] || {};
+    return get(field, 'name') || get(meta, 'primaryKeyFieldName');
```

Comment:
Did you consider adding a metadata computed property called `idFieldName` that would do:
```js
return this.getFieldsForTag('id')[0] || thi.get('primaryKeyFieldName')
```
That way we don't have to repeat this logic elsewhere in the future

Reply:
I did consider that, however, was concerned about blowing up with unnecessary computed properties for every tag that we may support in the future.  Since the look up was a one liner, I went with this direction for now.

Reply:
My concern is this logic isn't something that `dimension-bulk-import` should have to know about. If we are going to support a tag as a first class citizen, we should formally support it in the metadata model.

Reply:
Fixed